acceptanceTests/tap_test.go

```diff
@@ -10,10 +10,74 @@ import (
 	"os/exec"
 	"path"
 	"strings"
+	"sync"
 	"testing"
 	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/assert"
 )
 
+// waitTimeout waits for the waitgroup for the specified max timeout.
+// Returns true if waiting timed out.
+func waitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
+	c := make(chan struct{})
+	go func() {
+		defer close(c)
+		wg.Wait()
+	}()
+	select {
+	case <-c:
+		return false // completed normally
+	case <-time.After(timeout):
+		return true // timed out
+	}
+}
+
+// checkDBHasEntries checks whether there are any entries in the database
+// before the given timestamp. Returns a slice of non-empty entries if it succeeds.
+func checkDBHasEntries(t *testing.T, timestamp int64, limit int) (entries []map[string]interface{}) {
+	query := fmt.Sprintf("timestamp < %d and limit(%d)", timestamp, limit)
+	webSocketUrl := getWebSocketUrl(defaultApiServerPort)
+
+	c, _, err := websocket.DefaultDialer.Dial(webSocketUrl, nil)
+	assert.Nil(t, err)
+	defer c.Close()
+
+	handleWSConnection := func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		for {
+			_, message, err := c.ReadMessage()
+			if err != nil {
+				return
+			}
+
+			var data map[string]interface{}
+			err = json.Unmarshal([]byte(message), &data)
+			assert.Nil(t, err)
+
+			if data["messageType"] == "entry" {
+				entries = append(entries, data)
+			}
+		}
+	}
+
+	err = c.WriteMessage(websocket.TextMessage, []byte(query))
+	assert.Nil(t, err)
+
+	var wg sync.WaitGroup
+	go handleWSConnection(&wg)
+	wg.Add(1)
+
+	waitTimeout(&wg, 1*time.Second)
+
+	if len(entries) == 0 {
+		t.Error("unexpected entries result - Expected more than 0 entries")
+	}
+
+	return
```

Comment:
no need for return here

Reply:
It's a `missing return` compile-time error if we remove the `return` statement there.

Reply:
so why do you need to have a return value when you don't return the entries at all?

Reply:
`entries []map[string]interface{}` is defined in the method signature:

https://github.com/up9inc/mizu/blob/3e2102e262089f85b69c93b5ea25980b1f7e6824/acceptanceTests/testsUtils.go#L339

Populated on this line:

https://github.com/up9inc/mizu/blob/3e2102e262089f85b69c93b5ea25980b1f7e6824/acceptanceTests/testsUtils.go#L365

Checked on this line:

https://github.com/up9inc/mizu/blob/3e2102e262089f85b69c93b5ea25980b1f7e6824/acceptanceTests/testsUtils.go#L382

Returned on this line:

https://github.com/up9inc/mizu/blob/3e2102e262089f85b69c93b5ea25980b1f7e6824/acceptanceTests/testsUtils.go#L386

I don't understand what's the problem here?

Reply:
I see, the name of the function confused me. because you can't understand from the name of the function that it returns the entries (from the name i understood that it just checks if the db has entries)

Reply:
Is it resolved?

Reply:
IMO you should change the name of the function to getDBEntries and move the if len(entries) == 0 check to the tests/other function.

Reply:
Done with https://github.com/up9inc/mizu/pull/443/commits/9ff9be9d6804c8bf05a68f977aac17d5bf7e11f5